src/rgw/rgw_acl_swift.cc

```diff
@@ -75,6 +76,7 @@ static bool extract_referer_urlspec(const std::string& uid,
   url_spec = uid.substr(pos + 1);
 
   return sub.compare(".r") == 0 ||
+         sub.compare(".ref") == 0 ||
```

Comment:
Yup, [Swift does this](https://github.com/openstack/swift/blob/2.12.0/swift/common/middleware/acl.py#L102.) in exactly the same way.